runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -185,7 +186,7 @@ private StreamingDataflowWorker(
       StreamingCounters streamingCounters,
       MemoryMonitor memoryMonitor,
       AtomicInteger maxWorkItemCommitBytes,
-      GrpcWindmillStreamFactory windmillStreamFactory,
+      @Nullable GrpcWindmillStreamFactory windmillStreamFactory,
```

Comment:
why do we need both windmillServer and windmillStreamFactory here?

I see `windmillServer.appendSummaryHtml` in turn calls `windmillStreamFactory.appendSummaryHtml`, can we remove `windmillStreamFactory` from here?

Reply:
We need both since windmillStreamFactory is source of truth (it is used in `GrpcWindmillServer#appendSummaryHtml`)

In direct path mode we don't use `GrpcWindmillServer/WindmillServerStub` just inject windmillStreamFactory directly so will need that to add to status pages